eng/common/scripts/Invoke-GitHubAPI.ps1

```diff
@@ -392,28 +387,40 @@ function Update-GitHubIssue {
 
 function Remove-GitHubSourceReferences  {
   param (
-    [Parameter(Mandatory = $true)]
     $RepoOwner,
-    [Parameter(Mandatory = $true)]
     $RepoName,
+    $RepoId = "$RepoOwner/$RepoName",
     [ValidateNotNullOrEmpty()]
     [Parameter(Mandatory = $true)]
     $Ref,
     [ValidateNotNullOrEmpty()]
     [Parameter(Mandatory = $true)]
     $AuthToken
   )
-
   if ($Ref.Trim().Length -eq 0)
   {
     throw "You must supply a valid 'Ref' Parameter to 'Delete-GithubSourceReferences'."
   }
-
-  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/git/refs/$Ref"
+  $uri = "$GithubAPIBaseURI/$RepoId/git/refs/$Ref"
 
   return Invoke-RestMethod `
           -Method DELETE `
           -Uri $uri `
           -Headers (Get-GitHubApiHeaders -token $AuthToken) `
           -MaximumRetryCount 3
-}
\ No newline at end of file
+}
+
+
+function Get-GithubReferenceCommitDate($commitUrl) {
+  try {
+    $commitResponse = Invoke-RestMethod $commitUrl
```

Comment:
We should set MaximumRetryCount 3 here like other places. We might also need to pass the headers because if this is happening on a private repo we will need to use that token.